client/src/components/Schedule/TableView.tsx

```diff
@@ -0,0 +1,165 @@
+
+type Props = {
+  data: CourseEvent[];
+  timeZone: string;
+};
+
+export function TableView({ data, timeZone }: Props) {
+  return (
+    <Table
+      rowKey={record => (record.event.type === TaskTypes.deadline ? `${record.id}d` : record.id).toString()}
+      pagination={false}
+      dataSource={data}
+      // onRow={() => ({
+      //   onClick: ({ target }) => {
+      //     if (!target.closest('a')) {
+      //       // eslint-disable-next-line no-console
+      //       console.log('row is clicked');
+      //     }
+      //   },
+      // })}
+      columns={[
+        {
+          title: columnFilter(),
+          width: 20,
+          dataIndex: '#',
+          render: (_text, _record, index) => index + 1,
+        },
+        {
+          title: 'Date',
+          width: 120,
+          dataIndex: 'dateTime',
+          render: dateRenderer(timeZone),
+          sorter: dateSorter('dateTime'),
+          defaultSortOrder: 'descend',
+        },
+        { title: 'Time', width: 60, dataIndex: 'dateTime', render: timeRenderer(timeZone) },
+        {
+          title: 'Type',
+          width: 100,
+          dataIndex: ['event', 'type'],
+          render: (value: keyof typeof EventTypeColor) => (
+            <Tag color={EventTypeColor[value]}>{EventTypeToName[value] || value}</Tag>
+          ),
+        },
+        {
+          title: 'Special',
+          width: 150,
+          dataIndex: 'special',
+          render: tags => {
+            if (!tags) {
+              return;
+            }
+            return tagsRenderer(tags.split(','));
+          },
+        },
+        {
+          title: 'Name',
+          dataIndex: ['event', 'name'],
+          render: (value: string, record) => {
+            return record.event.descriptionUrl ? (
+              <a target="_blank" href={record.event.descriptionUrl}>
+                {value}
+              </a>
+            ) : (
+              value
+            );
+          },
+          ...getColumnSearchProps('event.name'),
+        },
+        {
+          title: 'Url',
+          width: 30,
+          dataIndex: 'broadcastUrl',
+          render: url => {
+            return url ? (
+              <Tooltip placement="topLeft" title={url}>
+                <a target="_blank" href={url}>
+                  {urlRenderer(url)}
+                </a>
+              </Tooltip>
+            ) : (
+              ''
+            );
+          },
+        },
+        { title: 'Duration', width: 60, dataIndex: 'duration' },
+        {
+          title: 'Organizer',
+          width: 140,
+          dataIndex: ['organizer', 'githubId'],
+          render: (value: string) => (value ? <GithubUserLink value={value} /> : ''),
+          ...getColumnSearchProps('organizer.githubId'),
+        },
+        {
+          title: 'Place',
+          dataIndex: 'place',
+          render: (value: string) => {
+            return value === 'Youtube Live' ? (
```

Comment:
let's move the render to it's own function

Reply:
done